app/src/main/java/com/kickstarter/libs/TrackingClientType.kt

```diff
@@ -0,0 +1,126 @@
+package com.kickstarter.libs
+
+import com.kickstarter.libs.utils.KoalaUtils
+import com.kickstarter.libs.utils.MapUtils
+import com.kickstarter.models.User
+import org.json.JSONArray
+import java.util.*
+
+abstract class TrackingClientType {
+    enum class Type(val tag: String) {
+        LAKE("💧 Lake"), SEGMENT("\uD83C\uDF81 Segment");
+    }
+
+    protected abstract val isGooglePlayServicesAvailable: Boolean
+    protected abstract val isTalkBackOn: Boolean
+
+    abstract fun optimizely(): ExperimentsClientType?
+    abstract fun loggedInUser(): User?
+
+    protected abstract fun type(): Type
```

Comment:
nit: could we alphabetize type? Seems like the rest of this list is in alphabetical order.

Reply:
Done